packages/react-native-codegen/src/parsers/typescript/components/props.js

```diff
@@ -9,26 +9,23 @@
  */
 
 'use strict';
-const {
-  flattenProperties,
-  getSchemaInfo,
-  getTypeAnnotation,
-} = require('./componentsUtils.js');
+const {getSchemaInfo, getTypeAnnotation} = require('./componentsUtils.js');
 
-import type {NamedShape, PropTypeAnnotation} from '../../../CodegenSchema.js';
+import type {
+  NamedShape,
+  StateTypeAnnotation,
+  PropTypeAnnotation,
+} from '../../../CodegenSchema.js';
 import type {TypeDeclarationMap} from '../../utils';
 
 // $FlowFixMe[unclear-type] there's no flowtype for ASTs
 type PropAST = Object;
 
-function buildPropSchema(
+function buildPropSchema<T>(
   property: PropAST,
   types: TypeDeclarationMap,
-): ?NamedShape<PropTypeAnnotation> {
+): NamedShape<T> {
```

Comment:
If it is a `propSchema`, why do you need to transform it in a Generic function? Does it support any kind of schema? If yes, we should change the name to `buildSchema`. What do you think?

Reply:
I have no better idea, this function is only for `PropTypeAnnotation` and `StateTypeAnnotation`, as a non-export function which is used right below, I think it is clear enough.

From my perspective, if properties and states are the say, we should not have `type StateTypeAnnotation = PropTypeAnnotation` from the beginning in `CodegenSchema.js`.